src/services/resource.js

```diff
@@ -81,10 +99,70 @@ module.exports = class resourceHelper {
 
 				const inReviewResources = await resourceQueries.findAll(filter, ['id'])
 
-				const uniqueInReviewResourcesIds = [...new Set(inReviewResources.map((item) => item.id))]
-
+				uniqueInReviewResourcesIds = utils.getUniqueElements(inReviewResources.map((item) => item.id))
+
+				const fetch_resource_ids_from_reviews = await reviewsQueries.findAll(
+					{
+						resource_id: {
+							[Op.in]: uniqueResourceIds,
+						},
+						status: {
+							[Op.in]: [
+								common.REVIEW_STATUS_INPROGRESS,
+								common.REVIEW_STATUS_REJECTED,
+								common.REVIEW_STATUS_REJECTED_AND_REPORTED,
+							],
+						},
+						organization_id: {
+							[Op.in]: OrganizationIds,
+						},
+					},
+					['resource_id', 'status', 'reviewer_id', 'notes', 'created_at', 'updated_at']
+				)
+
+				fetch_resource_ids_from_reviews.forEach((item) => {
+					if (item.status === common.REVIEW_STATUS_INPROGRESS) {
+						// Initialize the array if it doesn't exist
+						if (!in_progress_resource_ids[item.resource_id]) {
+							in_progress_resource_ids[item.resource_id] = []
+						}
+						// Add the details to the array
+						in_progress_resource_ids[item.resource_id].push({
+							resource_id: item.resource_id,
+							reviewer_id: item.reviewer_id,
+							notes: item.notes,
+							created_at: item.created_at,
+						})
+					} else if (item.status === common.REVIEW_STATUS_REJECTED) {
+						// Initialize the array if it doesn't exist
+						if (!rejected_resource_ids[item.resource_id]) {
+							rejected_resource_ids[item.resource_id] = []
+						}
+						// Add the details to the array
+						rejected_resource_ids[item.resource_id].push({
+							resource_id: item.resource_id,
+							reviewer_id: item.reviewer_id,
+							notes: item.notes,
+							rejected_at: item.updated_at,
+							review_status: common.REVIEW_STATUS_REJECTED,
+						})
+					} else if (item.status === common.REVIEW_STATUS_REJECTED_AND_REPORTED) {
+						// Initialize the array if it doesn't exist
+						if (!rejected_and_resported_resource_ids[item.resource_id]) {
+							rejected_and_resported_resource_ids[item.resource_id] = []
+						}
+						// Add the details to the array
+						rejected_and_resported_resource_ids[item.resource_id].push({
+							resource_id: item.resource_id,
+							reviewer_id: item.reviewer_id,
+							notes: item.notes,
+							rejected_at: item.updated_at,
+							review_status: common.REVIEW_STATUS_REJECTED_AND_REPORTED,
+						})
+					}
+				})
 				// count the number of resources with changes requested
-				const changesCount = await reviewsQueries.countDistinct({
+				const findAndCount = await reviewsQueries.countDistinct({
```

Comment:
@adithyadinesh0412 use find all and take the count and use this for the below function

Reply:
updated the query in countDistinct @priyanka-TL